libparsec/crates/platform_storage/src/workspace.rs

```diff
@@ -217,20 +224,35 @@ impl WorkspaceStorage {
     }
 
     pub async fn set_prevent_sync_pattern(&mut self, pattern: &Regex) -> anyhow::Result<bool> {
-        self.platform.set_prevent_sync_pattern(pattern).await
+        // TODO: Should we skip the next call if the pattern is the same?
+        let applied = self.platform.set_prevent_sync_pattern(pattern).await?;
+
+        self.prevent_sync_pattern = pattern.clone();
+        self.prevent_sync_pattern_applied = applied;
+
+        Ok(applied)
     }
 
-    pub async fn get_prevent_sync_pattern(&mut self) -> anyhow::Result<(Regex, bool)> {
-        self.platform.get_prevent_sync_pattern().await
+    pub fn get_prevent_sync_pattern(&self) -> (&Regex, bool) {
+        (
+            &self.prevent_sync_pattern,
+            self.prevent_sync_pattern_applied,
+        )
     }
 
     pub async fn mark_prevent_sync_pattern_fully_applied(
         &mut self,
         pattern: &Regex,
     ) -> anyhow::Result<bool> {
-        self.platform
+        // TODO: Should we skip the next call if the pattern is different?
+        let applied = self
```

Comment:
Returning `applied` and using it to set `self.prevent_sync_pattern_applied` seems a bad idea to me:

- If the pattern is different but fully applied, then we get `true` as result.
- Obviously `self.platform.mark_prevent_sync_pattern_fully_applied` is about setting `prevent_sync_pattern_applied` in the database to `true`, so why the method should ever return `false` ?

In the implementation of `platform.mark_prevent_sync_pattern_fully_applied` there is this:
https://github.com/Scille/parsec-cloud/blob/cc8383e117390d13fd915bb013f087f84835ba48/libparsec/crates/platform_storage/src/native/workspace.rs#L520-L525

I think returning the information if the is fully applied in the case the pattern is not the one we are expecting is the wrong approach here (if the pattern is not the one we are expecting, then something wrong is going on since we shouldn't be handling this kind concurrency at the database level, cf. the sync pattern cache field)

So I would change the signature of `self.platform.mark_prevent_sync_pattern_fully_applied` to also return the `MarkPreventSyncPatternFullyAppliedError` that allows to return the fact the provided `expected_pattern` is no the one in database.
And if this error occurs, then a panic should be raised (since the normal operation is to have this error being handled when checking the cache value of the prevent sync pattern)

Reply:
done